x-pack/legacy/plugins/ml/public/data_frame/pages/job_management/components/job_list/transform_table.ts

```diff
@@ -69,26 +69,29 @@ const ExpandableTable = (EuiInMemoryTable as any) as ComponentWithConstructor<Ex
 
 export class TransformTable extends ExpandableTable {
   static getDerivedStateFromProps(nextProps: any, prevState: any) {
+    const derivedState = { ...prevState.prevProps };
     if (nextProps.items !== prevState.prevProps.items) {
-      // We have new items because an external search has completed, so reset pagination state.
-      return {
+      Object.assign(derivedState, {
         prevProps: {
-          ...prevState.prevProps,
           items: nextProps.items,
         },
-        pageIndex: nextProps.pagination.initialPageIndex,
-      };
+      });
     }
+    Object.assign(derivedState, {
```

Comment:
looks like these could be added when `derivedState` is declared.

Reply:
👍 changed in d8094e0.